chain/chain/src/chain.rs

```diff
@@ -1144,6 +1145,88 @@ impl Chain {
         Ok(())
     }
 
+    /// This function validates the chunk_endorsements present in the block body. We verify the signature against
+    /// the ordered_chunk_validators for the shard.
+    /// Note that the chunk_validator_assignments as got as of the height_created for the chunk as it can potentially
+    /// be included in a later block.
+    fn validate_chunk_endorsements_in_block(&self, block: &Block) -> Result<(), Error> {
```

Comment:
Do you mind moving this out of chain.rs? I'm fine with having a file right next to it called chunk_endorsement.rs to contain this logic, like how chunk_validation.rs does it. We should avoid dumping things in chain.rs and making this giant file even larger.

Reply:
Yeah, makes sense... Will push outside this file but I still intend to keep it a part of chain.rs as it's called within chain.rs